rllib/core/learner/torch/torch_learner.py

```diff
@@ -630,6 +614,25 @@ def _map_module_to_device(self, module: MultiRLModule) -> None:
                 if isinstance(module[key], torch.nn.Module):
                     module[key].to(self._device)
 
+    def _compute_off_policyness(self, batch):
+        # Log off-policy'ness of this batch wrt the current weights.
+        off_policyness = {
+            (mid, DIFF_NUM_GRAD_UPDATES_VS_SAMPLER_POLICY): (
+                (self._weights_seq_no - module_batch[WEIGHTS_SEQ_NO]).float()
```

Comment:
not sure I understand the question.

We measure here the difference between a) number of updates the Learner has done thus far on the model vs b) number of updates the samplers' policies (on average) have undergone thus far. Since the samplers are always behind, this delta is always some small positive number. For PPO, it should actually be zero, for APPO/IMPALA somewhat < 100, etc..